SparkleSpin/Controllers/PlayerViewController.swift

```diff
@@ -11,51 +11,97 @@ import UIKit
 class PlayerViewController: UIViewController {
     
     @IBOutlet weak var playerTableView: UITableView!
+    @IBOutlet weak var playerEntryTextField: CustomTextField!
+    @IBOutlet weak var addPlayerButton: LightButton! 
+    
+    let playerViewModel = PlayerViewModel()
+    var doneBarButton: UIBarButtonItem?
     
-
     override func viewDidLoad() {
         super.viewDidLoad()
         setNavigationBarTitle()
         registerNib()
         setupTableView()
-    
+        addDoneBarButtonItem()
     }
     
     private func setNavigationBarTitle() {
-        navigationItem.title = "Choose Players!"
+        navigationItem.title = "Add Players!"
+    }
+    
+    private func addDoneBarButtonItem() {
+        doneBarButton = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: nil)
+        navigationItem.rightBarButtonItem = doneBarButton
+        doneBarButton?.isEnabled = false
     }
     
     private func setupTableView() {
-        playerTableView.dataSource = self
+        playerTableView.dataSource = playerViewModel
         playerTableView.delegate = self
         playerTableView.backgroundColor = ThemeColor.Light.primaryColor
         playerTableView.separatorStyle = .none
         playerTableView.rowHeight = UITableView.automaticDimension
         playerTableView.estimatedRowHeight = CGFloat(Constants.estimatedRowHeight)
         playerTableView.tableFooterView = UIView()
     }
-
-}
-
-extension PlayerViewController: UITableViewDataSource, UITableViewDelegate {
     
-    func registerNib() {
+    private func registerNib() {
         let entryCell = UINib(nibName: NibID.entryCell, bundle: nil)
         playerTableView.register(entryCell, forCellReuseIdentifier: CellID.entryCell)
     }
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+    private func addPlayerToList() {
+        let nameEntry = playerEntryTextField.text ?? ""
+        var playerToAdd = playerViewModel.createPlayerWith(name: nameEntry)
+        playerViewModel.addPlayerToSavedList(player: &playerToAdd)
```

Comment:
What was the reasoning to use `inout` for this function? And what do you think about changing lines 55 and 56 for something like this?

`playerViewModel.savePlayerWith(name: nameEntry)`

And then the implementation of `savePlayerWith(name: )` would be something like this: 

```
class PlayerViewModel: NSObject {
    func savePlayerWith(name: String) {
        let player = PlayerModel(name: name)
        playerList.append(player)
   }
}
```

I _think_ we don't need  the `createPlayerWith` function and also we don't need to create de model here, we can let the ViewModel to handle all those details for us. What do you think?

Reply:
> What was the reasoning to use `inout` for this function?

The player entries are currently identified with a `uuid` property, so I need to maintain the reference to the object itself (rather than just passing a copy) to maintain that entry's uniqueness. However, this a remnant of when I wasn't using the swipe to delete feature, and I needed a way to prevent entries with duplicate names from being deleted. So, in other words, it might not be needed anymore. I'll see if I can remove the `uuid` property without breaking anything. If so, I can pass a copy just fine.

> And what do you think about changing lines 55 and 56 for something like this? ...

Thanks! I'll consider this.

Reply:
The Player's `inout` property removed in https://github.com/BritneyS/SparkleSpin/pull/10/commits/95260b71d75ca4d0afdd685dbd319fdb03bf39ca

Reply:
Changes for Player creation (handled in viewmodel) in https://github.com/BritneyS/SparkleSpin/pull/10/commits/b386de87d709aeb02deeeb3b3d448d80d99daac3